File: 1701-average-waiting-time/1701-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double waitTime = 0;
        int currTime = 0;
        for (auto &x : customers) {
            int arrivalTime = x[0];
            int cookTime = x[1];
            if (currTime < arrivalTime) {
                currTime = arrivalTime;
            }
            int wait = currTime + cookTime - arrivalTime;
            waitTime += wait;
            currTime += cookTime;
        }
        int n = customers.size();
        return waitTime / n;
    }
};

File: 1701-average-waiting-time/1701-average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        n = len(customers)
        waitTime = 0
        currTime = 0
        for x in customers:
            arrival = x[0]
            cookTime = x[1]
            if currTime < arrival:
                currTime = arrival
            wait = currTime + cookTime - arrival
            currTime += cookTime
            waitTime += wait
        return waitTime / n